src/ws/service/friendship_event_handler.rs

```diff
@@ -0,0 +1,149 @@
+use std::sync::Arc;
+
+use crate::{
+    components::database::DatabaseComponentImplementation,
+    ws::{
+        app::SocialContext,
+        service::{
+            database_handler::{get_friendship, get_last_history, update_friendship_status},
+            errors::{FriendshipsServiceError, FriendshipsServiceErrorResponse},
+            types::{EventResponse, FriendshipPortsWs, RoomInfoWs},
+        },
+    },
+    UpdateFriendshipPayload,
+};
+
+use super::{
+    friendship_event_validator::validate_new_event,
+    friendship_status_calculator::get_new_friendship_status,
+    mapper::update_request_as_event_payload,
+    synapse_handler::{
+        get_or_create_synapse_room_id, set_account_data, store_message_in_synapse_room,
+        store_room_event_in_synapse_room,
+    },
+};
+
+/// Processes a friendship event update by validating it and updating the Database and Synapse.
+pub async fn handle_friendship_update(
+    request: UpdateFriendshipPayload,
+    context: Arc<SocialContext>,
+    acting_user: String,
+) -> Result<EventResponse, FriendshipsServiceErrorResponse> {
+    let event_payload = update_request_as_event_payload(request.clone())?;
+    let new_event = event_payload.friendship_event;
+    let second_user = event_payload.second_user;
+
+    let token = request
+        .auth_token
+        .ok_or(FriendshipsServiceError::InternalServerError)?
```

Comment:
Yes, u r right! but I deliberately mapped Internal Server Error because we haven't defined the list of errors yet. I have left a [TODO](https://github.com/decentraland/social-service/pull/186/files#diff-a6642970a18c36f1a743549cc8fec8e90d7eef262b568711fdc0f0a168a00896R5) to address these cases, which are a couple of them.